File: python/ex15.py
class SaldoInsuficienteError(Exception):
    def __init__(self, mensagem="Saldo insuficiente para a operação."):
        self.mensagem = mensagem
        super().__init__(self.mensagem)


class ContaCorrente:
    def __init__(self, titular_conta, saldo_inicial=0):
        self.titular_conta = titular_conta
        self.saldo = saldo_inicial

    def depositar(self, valor):
        self.saldo += valor
        print(f"Depósito de R${valor:.2f} realizado com sucesso. Saldo atual: R${self.saldo:.2f}")

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteError(f"Tentativa de saque: R${valor:.2f}. Saldo disponível: R${self.saldo:.2f}")
        self.saldo -= valor
        print(f"Saque de R${valor:.2f} realizado com sucesso. Saldo atual: R${self.saldo:.2f}")


def executar():
    conta_corrente = ContaCorrente("João", 100)

    try:
        conta_corrente.sacar(150)  
    except SaldoInsuficienteError as erro:
        print(erro)

    conta_corrente.depositar(50)  
    try:
        conta_corrente.sacar(100)  
    except SaldoInsuficienteError as erro:
        print(erro)

if __name__ == "__main__":
    executar()

File: java/ex15.java
public class ex15 {
    static class ExcecaoSaldoInsuficiente extends Exception {
        public ExcecaoSaldoInsuficiente(String mensagem) {
            super(mensagem);
        }
    }
    
    static class ContaCorrente {
        private double saldo;
        private String titular;
    
        public ContaCorrente(String titular, double saldoInicial) {
            this.titular = titular;
            this.saldo = saldoInicial;
        }
    
        public void depositar(double valor) {
            saldo += valor;
        }
    
        public void sacar(double valor) throws ExcecaoSaldoInsuficiente {
            if (valor > saldo) {
                throw new ExcecaoSaldoInsuficiente("Saldo insuficiente para realizar o saque.");
            }
            saldo -= valor;
        }
    
        public double obterSaldo() {
            return saldo;
        }
    
        public String obterTitular() {
            return titular;
        }
    }
    
    public class Main {
        public static void main(String[] args) {
            ContaCorrente conta = new ContaCorrente("João", 1000.00);
    
            try {
                conta.sacar(1500.00);
            } catch (ExcecaoSaldoInsuficiente e) {
                System.out.println(e.getMessage());
            }
    
            conta.depositar(500.00);
            try {
                conta.sacar(1500.00);
            } catch (ExcecaoSaldoInsuficiente e) {
                System.out.println(e.getMessage());
            }
    
            System.out.println("Saldo atual: R$ " + conta.obterSaldo());
        }
    }
    
}
